docs/tutorials/forum-application/7.vote-extension.md

```diff
@@ -0,0 +1,125 @@
+# Vote Extensions
+
+**In this section you will learn how a node can `ExtendVote` and `VerifyVote` on the forum application.**
+
+## ExtendVote
+
+The `ExtendVote` method allows applications to extend the pre-commit vote with arbitrary data. This allows applications
+to force their validators to do more than just validate blocks within consensus.
+
+When a validator is preparing to send a `pre-commit` vote, it first calls `ExtendVote.` The application then returns a
+blob of data called a `vote extension.` This data is opaque to the consensus algorithm but can contain application-specific
+information.
+
+The validator then sends both the `pre-commit` vote and the `vote extension` together to other validators. Other validators
+also call `ExtendVote` to generate their own `vote extensions.`
+
+When a validator receives a `pre-commit` vote with an attached `vote extension`, it calls `VerifyVoteExtension` to validate
+the `vote extension.` If valid, the validator includes the vote in its tally.
+
+The proposer of the next block will receive all `vote extensions` in `PrepareProposalRequest`.
+
+This allows validators to have access to all vote extensions at the next height. They can then use the data in the
+`vote extensions` to inform the transactions that make it into the next block.
+
+**Following is the code for the `ExtendVote` function:**
+
+```go
+// ExtendVote returns curse words as vote extensions
+func (app *ForumApp) ExtendVote(_ context.Context, _ *abci.ExtendVoteRequest) (*abci.ExtendVoteResponse, error) {
+	fmt.Println("Executing Application ExtendVote")
+
+	return &abci.ExtendVoteResponse{VoteExtension: []byte(app.CurseWords)}, nil
```

Comment:
How/where do you populate `app.CurseWords`?

Reply:
it's in the `abci/config.go` when you `LoadConfig` so the initial curse words are hard-coded